uwblueprint/headsupguys
# Tool Builder Linking and Publishing

## Notion ticket link

[[Tool Builder] Linking and Publishing Resolve Bugs](https://www.notion.so/uwblueprintexecs/Tool-Builder-Linking-and-Publishing-4c41f00dcb554c8daa01dd4bcd862518)



## Implementation description
- [x] Unpublish tool on save
- [x] Allow module linking without clicking into specific tool
- [x] Make Unpublish and link buttons work as intended
- [x] Don't allow publishing unless all fields are filled (check on publish click)
![image](https://user-images.githubusercontent.com/24593519/152700857-1add58b7-d84b-4b6c-9d72-a1f219ec898b.png)
 ![image](https://user-images.githubusercontent.com/24593519/152700871-2abf58c2-7280-44b2-95ce-b47a7d09f256.png)




## Steps to test
1. Make 2 new tools
2. Go back to the admin/dashboard/tools page and ensure that you can link a module to the first tool (you will need at least 1 module not currently linked to a tool)
3. Make sure you are unable to link the second tool to the same module
4. Fill out all the required fields on one of you tools and try publishing it from the admin/dashboard/tools page
5. Try unpublishing said tool from the same page
6. Try unlink the tool
7. Repeat steps 2 - 6 in random orders and ensure that no unexpected bugs occur



## What should reviewers focus on?
* Module unlinking, linking, publishing, unpublishing

## Checklist
